docs/extended-cpts.md

```diff
@@ -0,0 +1,312 @@
+# Extended Custom Post Types
+
+For the full documentation, refer to the [Extended CPTs wiki](https://github.com/johnbillion/extended-cpts/wiki).
+
+## Registering Post Types
+
+The `register_extended_post_type()` function is the core of Extended CPTs. It's a wrapper for WordPress' own `register_post_type()` function, which means that any parameters which are accepted by `register_post_type()` are accepted.
+
+The function's signature looks like this:
+
+```php
+function register_extended_post_type( string $post_type, array $args = [], array $names = [] ) : Extended_CPT
+```
+
+Need a simple post type with no frills? You can register a post type with a single parameter:
+
+```php
+add_action( 'init', function() {
+
+	register_extended_post_type( 'article' );
+
+} );
+```
+
+Try it. You'll have a hierarchical public post type with an admin UI, and all the labels and post updated messages will be automatically generated.
+
+Arguments go into the second parameter, and the third parameter is used to override the singular, plural, and slug arguments which are used as the basis for the post type labels, post updated messages, and permalinks.
+
+```php
+add_action( 'init', function() {
+
+	register_extended_post_type( 'story', [
+		# Add the post type to the site's main RSS feed:
+		'show_in_feed' => true,
+
+		# Show all posts on the post type archive:
+		'archive' => [
+			'nopaging' => true
+		],
+
+		# Add some custom columns to the admin screen:
+		'admin_cols' => [
+			'featured_image' => [
+				'title'          => 'Illustration',
+				'featured_image' => 'thumbnail'
+			],
+			'published' => [
+				'title'       => 'Published',
+				'meta_key'    => 'published_date',
+				'date_format' => 'd/m/Y'
+			],
+			'genre' => [
+				'taxonomy' => 'genre'
+			]
+		],
+
+		# Add a dropdown filter to the admin screen:
+		'admin_filters' => [
+			'genre' => [
+				'taxonomy' => 'genre'
+			]
+		]
+	],
+	[
+		# Override the base names used for labels:
+		'singular' => 'Story',
+		'plural'   => 'Stories',
+		'slug'     => 'stories'
+	] );
+
+} );
+```
+
+And that easily, we have a 'Stories' post type, with correctly generated labels and post updated messages, three custom columns in the admin area (two of which are sortable), stories added to the main RSS feed, and all stories displayed on the post type archive.
+
+## Registering taxonomies
```

Comment:
```suggestion
## Registering Taxonomies
```

We've standardised on title casing for the rest of the docs, just gonna add commit suggests to make this one consistent.